app/_src/guides/upgrading-transparent-proxy.md

```diff
@@ -0,0 +1,54 @@
+content_type: tutorial
+---
+
+{% capture docs %}/docs/{{ page.release }}{% endcapture %}
+{% assign Kuma = site.mesh_product_name %}
+
+{% capture Warning %}{% if page.edition and page.edition != "kuma" %}**Warning:** {% endif %}{% endcapture %}
+{% capture Important %}{% if page.edition and page.edition != "kuma" %}**Important:** {% endif %}{% endcapture %}
+{% capture brbr %}{% if page.edition and page.edition != "kuma" %}<br /><br />{% endif %}{% endcapture %}
+
+The core `iptables` rules applied by {{ Kuma }}'s transparent proxy rarely change, but occasionally new features may require updates. To upgrade the transparent proxy on Universal environments, follow these steps:
+
+### Step 1: Cleanup existing iptables rules{% if_version gte:2.9.x inline:true %} (conditional){% endif_version %}
+
+{% if_version gte:2.9.x %}
+{% warning %}
+{{ Important }}If you're upgrading from {{ Kuma }} version 2.9 or later, and you have **not** manually disabled the automatic addition of comments by setting `comments.disabled` to `true` in the transparent proxy configuration, **this step is unnecessary**.
+{{ brbr }}
+Starting with {{ Kuma }} 2.9, all `iptables` rules are tagged with comments, allowing {{ Kuma }} to track rule ownership. This enables `kumactl` to automatically clean up any existing `iptables` rules or custom chains created by previous versions of the transparent proxy. This process runs automatically at the start of the installation, eliminating the need for any manual cleanup beforehand.
+{% endwarning %}
+{% endif_version %}
+
+To manually remove existing `iptables` rules, you can either restart the host (if the rules were not persisted using system start-up scripts or `firewalld`), or run the following commands:
+
+{% danger %}
+{{ Warning }}These commands will remove **all** `iptables` rules and **all** custom chains in the specified tables, including those created by {{ Kuma }} as well as any other applications or services.
+{% enddanger %}
+
+```sh
+iptables --table nat --flush         # Flush all rules in the nat table (IPv4)
+ip6tables --table nat --flush        # Flush all rules in the nat table (IPv6)
+iptables --table nat --delete-chain  # Delete all custom chains in the nat table (IPv4)
+ip6tables --table nat --delete-chain # Delete all custom chains in the nat table (IPv6)
+
+# The raw table contains rules for DNS traffic redirection
+iptables --table raw --flush         # Flush all rules in the raw table (IPv4)
+ip6tables --table raw --flush        # Flush all rules in the raw table (IPv6)
+
+# The mangle table contains rules to drop invalid packets
+iptables --table mangle --flush      # Flush all rules in the mangle table (IPv4)
+ip6tables --table mangle --flush     # Flush all rules in the mangle table (IPv6)
+```
+
+### Step 2: Install new transparent proxy
+
+After clearing the `iptables` rules (if necessary), reinstall the transparent proxy by running:
+
+```sh
+kumactl install transparent-proxy [...]
```

Comment:
I'm not sure here, but I was thinking if we're putting this into `guides,` then every command has to be executable so that users can follow along. I think so far, we're following this principle with guides.